### PR TITLE
Bmb/current index

### PR DESCRIPTION
**Pull request by biyeun**
fyi for @wpride

### CODE REVIEW

**Comment by biyeun** [comment]
@benrudolph this should be g2g tests should be fixed and basic repeat groups are now supported

**Comment by biyeun** [comment]
@benrudolph merge along with https://github.com/dimagi/commcare-core/pull/423

File: src/main/java/objects/SerializableFormSession.java [hunk 1]
```diff
@@ -25,7 +25,7 @@
     private String title;
     private String dateOpened;
     private boolean oneQuestionPerScreen;
-    private int currentIndex;
+    private String currentIndex = "0";
```
Comment (benrudolph) [inline src/main/java/objects/SerializableFormSession.java]: should default here be `-1`?

File: src/main/java/session/FormSession.java [hunk 2]
```diff
@@ -311,23 +311,44 @@ public boolean getIsAtLastIndex() {
...
     public void stepToPreviousIndex() {
-        this.formEntryController.jumpToIndex(JsonActionUtils.indexFromString("" + currentIndex, formDef));
+        this.formEntryController.jumpToIndex(JsonActionUtils.indexFromString(currentIndex, formDef));
         FormEntryNavigator formEntryNavigator = new FormEntryNavigator(formEntryController);
         FormIndex newIndex = formEntryNavigator.getPreviousFormIndex();
+
+        // check if this index is the beginning of a group that is not a question list.
+        IFormElement element = formEntryController.getModel().getForm().getChild(newIndex);
+        while (element instanceof GroupDef && !formEntryController.isFieldListHost(newIndex)) {
+            log.info("step back group");
```
Comment (benrudolph) [inline src/main/java/session/FormSession.java]: this log message isn't very useful

**Comment by benrudolph** [comment]
test failure likely real

File: src/main/java/session/FormSession.java (hunk 2 above)
Comment (biyeun) [inline src/main/java/session/FormSession.java]: lol oops. thanks

File: src/main/java/objects/SerializableFormSession.java (hunk 1 above)
Comment (biyeun) [inline src/main/java/objects/SerializableFormSession.java]: updated

**Comment by biyeun** [comment]
@benrudolph ✅

**Pull request merged**